### DISCUSSION TITLE
pytorch3d and cuda12.1 not working together

### DISCUSSION THREAD

**Discussion by shayansep** [facebookresearch/pytorch3d#1821] [General]
python 3.10.10, CUDA=12.1
conda install pytorch==2.1.0 torchvision==0.16.0 torchaudio==2.1.0 pytorch-cuda=12.1 -c pytorch -c nvidia

If i write the versions of pytorch and torchvision my CUDA will become "True" in my python test code but, the pytorch3d installation will be incomplete with an error. If i dont write the versions it will install pytorch 2.3.1 and the pytorch3d installation will become complete but the CUDA will become "False"

(i install pytorch3d using these codes on minianaconda prompt window
git clone https://github.com/facebookresearch/pytorch3d.git
cd pytorch3d
python setup.py install)

on some verisons CUDA goes true and pytorch3d wont install and on some, the other way. is there a version that can work on both ???
im going crazy, i have uninstalled and installed many different versions 100 times during past 2 days.
*my nvidia CUDA is also 12.1 now ( using nvcc --version and nvidia-smi, both cudas are 12.1)*
(This is the pytorch3d error at the end of the installation: RuntimeError: Error compiling objects for extension)

please help me
thank you for your attention.

**Comment by rohit7044** [comment]
Have you considered using CUDA-11.8?
In my case it helped with my installation.
Give it a try

**Reply by shayansep** [reply]
i also tested it on CUDA 11.8
the same problem

**Comment by LucasColas** [comment]
Did you find a solution ?
